handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java

```diff
@@ -40,94 +34,24 @@
 import java.util.List;
 import java.util.Set;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
 import javax.net.ssl.SSLEngine;
 import javax.net.ssl.SSLException;
 import javax.net.ssl.SSLHandshakeException;
 
-import org.junit.After;
-import org.junit.Before;
 import org.junit.Test;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Mock;
-import org.mockito.MockitoAnnotations;
 
-public class JdkSslEngineTest {
+public class JdkSslEngineTest extends SSLEngineTest {
     private static final String PREFERRED_APPLICATION_LEVEL_PROTOCOL = "my-protocol-http2";
     private static final String FALLBACK_APPLICATION_LEVEL_PROTOCOL = "my-protocol-http1_1";
     private static final String APPLICATION_LEVEL_PROTOCOL_NOT_COMPATIBLE = "my-protocol-FOO";
 
-    @Mock
-    private MessageReciever serverReceiver;
-    @Mock
-    private MessageReciever clientReceiver;
-
-    private Throwable serverException;
-    private Throwable clientException;
-    private SslContext serverSslCtx;
-    private SslContext clientSslCtx;
-    private ServerBootstrap sb;
-    private Bootstrap cb;
-    private Channel serverChannel;
-    private Channel serverConnectedChannel;
-    private Channel clientChannel;
-    private CountDownLatch serverLatch;
-    private CountDownLatch clientLatch;
-
-    private interface MessageReciever {
-        void messageReceived(ByteBuf msg);
-    }
-
-    private final class MessageDelegatorChannelHandler extends SimpleChannelInboundHandler<ByteBuf> {
-        private final MessageReciever receiver;
-        private final CountDownLatch latch;
-
-        public MessageDelegatorChannelHandler(MessageReciever receiver, CountDownLatch latch) {
-            super(false);
-            this.receiver = receiver;
-            this.latch = latch;
-        }
-
-        @Override
-        protected void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
-            receiver.messageReceived(msg);
-            latch.countDown();
-        }
-    }
-
-    @Before
-    public void setup() {
-        MockitoAnnotations.initMocks(this);
-        serverLatch = new CountDownLatch(1);
-        clientLatch = new CountDownLatch(1);
-    }
-
-    @After
-    public void tearDown() throws InterruptedException {
-        if (serverChannel != null) {
-            serverChannel.close().sync();
-            Future<?> serverGroup = sb.group().shutdownGracefully(0, 0, TimeUnit.MILLISECONDS);
-            Future<?> serverChildGroup = sb.childGroup().shutdownGracefully(0, 0, TimeUnit.MILLISECONDS);
-            Future<?> clientGroup = cb.group().shutdownGracefully(0, 0, TimeUnit.MILLISECONDS);
-            serverGroup.sync();
-            serverChildGroup.sync();
-            clientGroup.sync();
-        }
-        clientChannel = null;
-        serverChannel = null;
-        serverConnectedChannel = null;
-        serverException = null;
-    }
-
     @Test
     public void testNpn() throws Exception {
```

Comment:
I will follow up with NPN and ALPN tests for openssl with another PR as it is not related to this feature.